src/components/TokenExpired.js

```diff
@@ -1,25 +1,91 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
+import { SessionTimeout } from 'patternfly-react'
 
 import ErrorAlert from './ErrorAlert'
 import { msg } from '_/intl'
 
+import { logout } from '_/actions'
+
+const MAX_INACTIVE_TIME = 60 * 20 // 20 minutes
+const TIME_TO_DISPLAY_MODAL = 30 // 30 seconds
+
 // TODO: allow the user to cancel the automatic reload?
 // If so, change config.isTokenExpired to false and add additional check to doCheckTokenExpired() before actual reload
-const TokenExpired = ({ config }) => {
-  if (!config.get('isTokenExpired')) {
-    return null
+class TokenExpired extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expired: false,
+      counter: MAX_INACTIVE_TIME,
+    }
+
+    this.dropCounter = this.dropCounter.bind(this)
+    this.decrementCounter = this.decrementCounter.bind(this)
+
+    document.body.addEventListener('mousemove', this.dropCounter)
```

Comment:
Is moving the mouse over the document considered interacting with the App?  That is inconsistent with session timeout counters on banking apps etc.  I would prefer requiring actual interaction with the app, for example click buttons/links/navigation or keyboard navigation or keyboard input.

@gregsheremeta, @lwrigh Do you have/know of any guide lines from UXD group around this?

Reply:
The problem with clicking is that some event handlers can `stopPropagation`, so document handler never get event. Mouse move means that portal tab is now active (not somewhere in background), so I think it is better solution for now.

Reply:
Ok, I'd still like to hear from @lwrigh and @gregsheremeta about which approach makes sense from the UX side before we decide for certain.

Reply:
yeah, I think clicks make more sense.

So for the things that stopPropagation, there's gotta be a hook somewhere?

Reply:
@bond95 , @gregsheremeta - I did a bit more poking around in non-react world (https://github.com/shawnmclean/Idle.js, https://github.com/kidh0/jquery.idle) to see what other people have done.  It's all similar for key events and mouse moves etc.  I'm still not fond of pure mouse motion tracking.

Other solutions attached listeners to `window` instead of `document.body`.  I'm not sure if they are subject to `e.preventDefault()` event propagation cancellation, but it is worth checking out.

Reply:
I think other products require the user to move their move across or scroll up and down the page to keep it active. From a UX perspective I would consider that to be a pretty standard practice for keeping a page active.